tensorrt_llm/_torch/auto_deploy/custom_ops/flashinfer_attention.py

```diff
@@ -125,30 +150,45 @@ def _plan_decode(wrapper: flashinfer.BatchDecodeWithPagedKVCacheWrapper):
             wrapper._paged_kv_last_page_len_buf.copy_(kv_last_page_len)
             return wrapper
 
-        # check for re-planning
-        if plan_params != self.plan_params:
-            if plan_params.is_generate:
-                _plan_decode(self.decode_wrapper)
-            else:
-                # plan prefill
-                self.prefill_wrapper.plan(
-                    qo_indptr,
-                    kv_page_indptr,
-                    kv_page_indices,
-                    kv_last_page_len,
-                    plan_params.n_heads,  # Q heads
-                    plan_params.n_kv_heads,  # KV heads
-                    plan_params.head_dim,
-                    plan_params.page_size,
-                    causal=plan_params.causal,
-                    q_data_type=plan_params.q_dtype,
-                    kv_data_type=plan_params.kv_dtype,
-                    sm_scale=plan_params.sm_scale,
+        # handle decode
+        if plan_params.is_generate:
+            wrapper = self.decode_wrapper
+            # Update internal buffers in-place (constant time)
+            wrapper._paged_kv_indptr_buf.copy_(kv_page_indptr)
+            wrapper._paged_kv_indices_buf[: len(kv_page_indices)].copy_(kv_page_indices)
+            wrapper._paged_kv_last_page_len_buf.copy_(kv_last_page_len)
```

Comment:
Do we really feel comfortable going through private atttributes of flashinfer? :O